specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/stable/2020-05-15/timeseriesinsights.json

```diff
@@ -0,0 +1,2549 @@
+            "$ref": "./examples/Operation_List.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully listed the available operations.",
+            "schema": {
+              "$ref": "#/definitions/OperationListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.TimeSeriesInsights/environments/{environmentName}": {
+      "put": {
+        "tags": [
+          "Environments"
+        ],
+        "operationId": "Environments_CreateOrUpdate",
+        "x-ms-examples": {
+          "EnvironmentsCreate": {
+            "$ref": "./examples/EnvironmentsCreate.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "description": "Create or update an environment in the specified subscription and resource group.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "environmentName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[-\\w\\._\\(\\)]+$",
+            "minLength": 1,
+            "maxLength": 90,
+            "description": "Name of the environment"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/EnvironmentCreateOrUpdateParameters"
+            },
+            "description": "Parameters for creating an environment resource."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The existing environment definition was successfully updated.",
+            "schema": {
+              "$ref": "#/definitions/EnvironmentResource"
+            }
+          },
+          "201": {
+            "description": "The environment create request was accepted. Environment provisioning is an asynchronous operation. You can periodically get your environment definition and monitor progress via the provisioningState property.",
+            "schema": {
+              "$ref": "#/definitions/EnvironmentResource"
+            }
+          },
+          "404": {
+            "description": "The subscription or resource group could not be found."
+          },
+          "default": {
+            "description": "HTTP 400 (Bad Request): The given environment request body is invalid; See the error code and message in the response for details.",
```

Comment:
>HTTP 400 (Bad Request):  [](start = 28, length = 24)

The `default` response is a catch-all for all kinds of errors (4xx and 5xx), so this is somewhat misleading.

Reply:
This applies to all occurrences of this pattern in this swagger.

---
In reply to: [400578474](https://github.com/Azure/azure-rest-api-specs/pull/8791#discussion_r400578474) [](ancestors = 400578474)